l10n_fr_payment_payfip/tests/test_payfip.py

```diff
@@ -0,0 +1,156 @@
+        self.addCleanup(self.PaymentTransaction._revert_method,
+                        "_handle_feedback_data")
+
+    def test_return_get(self):
+        idop = "5e64f6f2-7b4b-4ebe-aa6c-7493f5e443af"
+        url = self._build_url(PayFIPController._return_url)
+        response = self.opener.get(
+            url + "?idop=" + idop)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(self.calls, 1)
+        response = self.opener.post(
+            url, data={"idop": idop})
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(self.calls, 2)
+
+    def test_notify_post(self):
+        idop = "5e64f6f2-7b4b-4ebe-aa6c-7493f5e443af"
+        url = self._build_url(PayFIPController._notification_url)
+        response = self.opener.post(
+            url, data={"idop": idop})
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(self.calls, 1)
+
+
+@tagged('post_install', '-at_install')
+class PayFIPTest(PayFIPCommon):
+
+    def test_processing_values(self):
+        tx = self.create_transaction(flow='redirect')  # Only flow implemented
+        return_url = self._build_url(PayFIPController._return_url)
+        notification_url = self._build_url(PayFIPController._notification_url)
+
+        expected_values = {
+            'numcli': self.acquirer.payfip_customer_number,
+            'exer': str(fields.Datetime.now().year),
+            'mel': 'norbert.buyer@example.com',
+            'montant': "1111.11",
+            'objet': self.reference,
+            'refdet': tx.acquirer_reference,
+            'saisie': 'T',
+            'urlnotif': notification_url,
+            'urlredirect': return_url
+        }
+
+        with mute_logger('odoo.addons.payment.models.payment_transaction'):
+            processing_values = tx._get_processing_values()
+        redirect_form_data = self._extract_values_from_html_form(
+            processing_values['redirect_form_html'])
+
+        self.assertEqual(
+            redirect_form_data['action'], self.acquirer.payfip_get_form_action_url())
+
+        self.assertDictEqual(
+            expected_values,
+            redirect_form_data['inputs'],
+            "PayFIP: invalid inputs specified in the redirect form.",
+        )
+
+    def test_feedback_processing(self):
+        # typical data posted by payFIP after client has successfully paid
+        payfip_post_data = {
+            'dattrans': u'06012023',
+            'exer': u'2023',
+            'heurtrans': u'1100',
+            'idOp': u'93be8501-9184-4e63-81b3-53b5a7f4d69a',
+            'mel': u'norbert.buyer@example.com',
+            'montant': u'111111',
+            'numauto': u'A55A',
+            'numcli': self.acquirer.payfip_customer_number,
+            'objet': self.reference,
+            'refdet': u'088655675121650',
+            'saisie': u'T'
+        }
+
+        with self.assertRaises(ValidationError):
```

Comment:
Tips: it's a good practice to prefer assertRaisesRegex and make sure the expected message has been raised, in some circonstance as in odoo we often raise the same set of exception we can be surprised an other module or part of the code is raising the exception.